source/common/http/conn_pool_grid.cc

```diff
@@ -316,21 +330,33 @@ void ConnectivityGrid::markHttp3Broken() { http3_status_tracker_.markHttp3Broken
 
 void ConnectivityGrid::markHttp3Confirmed() { http3_status_tracker_.markHttp3Confirmed(); }
 
-void ConnectivityGrid::onDrainReceived() {
+bool ConnectivityGrid::isIdle() const {
+  // This is O(n) but the function is constant and there are no plans for n > 8.
+  bool idle = true;
+  for (const auto& pool : pools_) {
+    idle &= pool->isIdle();
+  }
+  return idle;
+}
```

Comment:
I had something like that, and it made test expectations annoyingly difficult to read, so I wanted to call `isIdle()` on all the pools (all being 2-3).

I have mixed feelings on bitwise operators on booleans; the suggestion came from @alyssawilk in the comments on the previous PR.

The probably "correct" version is to use `std::all_of`, but the syntax isn't very nice; it would still cause issues with my test expectations.